common/client/multi_node.go

```diff
@@ -265,22 +264,37 @@ func (c *multiNode[CHAIN_ID, SEQ, ADDR, BLOCK_HASH, TX, TX_HASH, EVENT, EVENT_OP
 	return c.activeNode, err
 }
 
-// nLiveNodes returns the number of currently alive nodes, as well as the highest block number and greatest total difficulty.
-// totalDifficulty will be 0 if all nodes return nil.
-func (c *multiNode[CHAIN_ID, SEQ, ADDR, BLOCK_HASH, TX, TX_HASH, EVENT, EVENT_OPS, TX_RECEIPT, FEE, HEAD, RPC_CLIENT, BATCH_ELEM]) nLiveNodes() (nLiveNodes int, blockNumber int64, totalDifficulty *big.Int) {
-	totalDifficulty = big.NewInt(0)
+// LatestChainInfo - returns number of live nodes available in the pool, so we can prevent the last alive node in a pool from being marked as out-of-sync.
+// Return highest ChainInfo most recently received by the alive nodes.
+// E.g. If Node A's the most recent block is 10 and highest 15 and for Node B it's - 12 and 14. This method will return 12.
+func (c *multiNode[CHAIN_ID, SEQ, ADDR, BLOCK_HASH, TX, TX_HASH, EVENT, EVENT_OPS, TX_RECEIPT, FEE, HEAD, RPC_CLIENT, BATCH_ELEM]) LatestChainInfo() (int, ChainInfo) {
+	var nLiveNodes int
+	ch := ChainInfo{
+		TotalDifficulty: big.NewInt(0),
+	}
 	for _, n := range c.nodes {
-		if s, num, td := n.StateAndLatest(); s == nodeStateAlive {
+		if s, nodeChainInfo := n.StateAndLatest(); s == nodeStateAlive {
 			nLiveNodes++
-			if num > blockNumber {
-				blockNumber = num
-			}
-			if td != nil && td.Cmp(totalDifficulty) > 0 {
-				totalDifficulty = td
-			}
+			ch.BlockNumber = max(ch.BlockNumber, nodeChainInfo.BlockNumber)
+			ch.FinalizedBlockNumber = max(ch.FinalizedBlockNumber, nodeChainInfo.FinalizedBlockNumber)
+			ch.SetTotalDifficultyIfGt(nodeChainInfo.TotalDifficulty)
 		}
 	}
-	return
+	return nLiveNodes, ch
+}
+
+// HighestUserObservations - returns highest ChainInfo ever observed by any user of the MultiNode
+func (c *multiNode[CHAIN_ID, SEQ, ADDR, BLOCK_HASH, TX, TX_HASH, EVENT, EVENT_OPS, TX_RECEIPT, FEE, HEAD, RPC_CLIENT, BATCH_ELEM]) HighestUserObservations() ChainInfo {
```

Comment:
I find it hard to distinguish this from the node's `HighestUserObservations` since they have the same name and also the notion of `user` doesn't really make sense to me. How about `MultiNodeHighestObesrvations` and `NodeHighestObservation` for the node's method? I'm open to other suggestions as well, it's just that `user` usually refers to some person that actually triggers API calls.

Reply:
> it's just that user usually refers to some person that actually triggers API calls.

That's exactly the same reference we use here. Our goal with `HighestUserObservations` is to communicate what was observed by the user/application layer, excluding the health check results. [Here](https://github.com/smartcontractkit/chainlink/pull/13048#discussion_r1643105065) are more details on why we want to do it.

What do you think of `HighestAppObservations` or `HighestExternalObservations`?
I'm not sure that we need to distinguish MultiNode's and Node's methods, as they are doing exactly the same thing, just on different levels.